src/mlpack/methods/ann/image_functions/bilinear_function.hpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file bilinear_function.hpp
+ * @author Kris Singh
+ *
+ * Definition and implementation of the bilinear interpolation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_INTERPOLATION_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_INTERPOLATION_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The bilinear interpolation function
+ *
+ */
+class BiLinearFunction
+{
+ public:
+  BiLinearFunction(
+      const size_t inRowSize,
+      const size_t inColSize,
+      const size_t scale):
+      inRowSize(inRowSize),
+      inColSize(inColSize),
+      outRowSize(2 * inRowSize),
+      outColSize(2 * inColSize),
+      scale(scale)
+  {};
+  /**
+   * UpSample the given Image
+   *
+   * @param input Input image.
+   * @param output The resulting interpolated output image.
+   */
+  template<typename eT>
+  void UpSample(const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  {
+    // Get dimensions.
+    arma::Mat<eT> tempInput = input;
```

Comment:
I wanted to use the case of 1d array as input to the bilinear function. If I use 1d array I would have to resize to the `inRowSize, inColSize` which will not be possible since I am using const for input. That why I used a temp variable here.